### DISCUSSION TITLE
TranslationTransform2D Exact X-Y Shift

### DISCUSSION THREAD

**Discussion by aditsanghvi94** [dipy/dipy#2354] [Registration] [answered]
## Description
How do I get the shift in pixels between two images using TranslationTransform2D?
I have two (128,512) images that I want to compare - static and moving. This is the code I am using 
```
    transform = TranslationTransform2D()
    params0 = None
    translation = affreg.optimize(static, moving, transform, params0)
    starting_affine = translation.affine
    warped_moving = translation.transform(moving)
    
```

How do I interpret the output of translation.affine? Is the 1st row, 3rd column = shift in x pixels and 2nd row,3rd column = shift in y pixels?
e.g. I just want to say that moving is static , but with a shift of +10 in x-dimension and +20 in y dimension

**Comment by skoudoro** [comment, marked as answer]
Hi @aditsanghvi94,

Sorry for the late reply but yes, this is the interpretation.

Maybe we should create a utility function Also, you can do:

```python
>>> print("{:translation}".format(translation))
>>> print("{:rotation}".format(translation))
```

I am closing this issue. feel free to reopen it if needed
